cmd/earthly/build_cmd.go

```diff
@@ -225,14 +225,14 @@ func (app *earthlyApp) actionBuildImp(cliCtx *cli.Context, flagArgs, nonFlagArgs
 	app.console.PrintPhaseHeader(builder.PhaseInit, false, "")
 	app.warnIfArgContainsBuildArg(flagArgs)
 
-	err = app.initFrontend(cliCtx)
+	err = app.configureSatellite(cliCtx, cloudClient, gitCommitAuthor, gitConfigEmail)
 	if err != nil {
-		return err
+		return errors.Wrapf(err, "could not construct new buildkit client")
 	}
 
-	err = app.configureSatellite(cliCtx, cloudClient, gitCommitAuthor, gitConfigEmail)
+	err = app.initFrontend(cliCtx)
```

Comment:
I changed the calling order here because initFrontend sets the majority of buildkitsettings properties according to a configueration that is not up to date.
More specifically, useTCP is determined according to the bkURL but in case of a satellite, the satellite host is only set in `configureSatellite`, so without changing the order, the value of UseTCP was set to false, which is wrong.